### ISSUE TITLE
NSRangeException on  XWVHttpConnection.swift line 108

### ISSUE THREAD

**Issue by prep4gmat** [XWebView/XWebView#35]
I am loading HTML string with MathJax if it s relevant in any way.
In some cases I crash with this exception - 

```
Fatal Exception: NSRangeException
*** -[NSConcreteMutableData subdataWithRange:]: range {0, 331} exceeds data length 329
```

Looks like the buffer length is smaller then the data range.

Adding this code -

`let inputBufferLength = inputBuffer.length
      var range = NSRange(bytesConsumed...base.distanceTo(ptr))
                    if range.length > inputBufferLength {
                        print("oops")
                        range.length = inputBufferLength
                    }
                    let data = inputBuffer.subdataWithRange(range)`

Fixes the issue but I admit I do not understand the root of the issue.

Thanks
Shani

**Comment by xwv** [comment]
Is this issue reproducible? How often it occurs?
Can you identify a pattern?

If convenient, could you please provide more information, like:
1. The value of `cursor` variable once the exception is raised.
2. The full content of the HTTP request.

I have a try (not tested) for this issue:

```
            var bytesConsumed = 0
            var ptr = cursor > 3 ? base.advancedBy(cursor - 3): base
-           for _ in 0 ..< bytesReaded {
+           while base.distanceTo(ptr) < bytesReaded - 3 {
                if UnsafePointer<UInt32>(ptr).memory == UInt32(bigEndian: 0x0d0a0d0a) {
                    // End of request header.
```

**Comment by xwv** [comment]
@prep4gmat , any update?

**Comment by yeti23** [comment]
We use XWebView on iOS and OS X (easy port). We have this error only on OS X.

2016-04-06 09:26:02.824 EXAOCampusX[24800:2374861] An uncaught exception was raised
2016-04-06 09:26:02.824 EXAOCampusX[24800:2374861] **\* -[NSConcreteMutableData subdataWithRange:]: range {0, 310} exceeds data length 308

Adding this code to fix it.
if base.distanceTo(ptr) < inputBuffer.length {
...

**Comment by xwv** [comment]
@yeti23, have you identified the root cause?
If yes, could you please submit a PR for the bug?
Meanwhile, I would like to pull your OS X porting.

**Comment by yeti23** [comment]
I haven't identifier the problem. I join the iOS/OSX version.
You can easy reproduce the error on OS X. Just remove my comment in 
XWVHttpConnection.

I have added an http connexion on iOS 9, because we needed an ajax 
called. Else we get a CORS error.

**Comment by xwv** [comment]
@yeti23 , I think my patch can fix it. Could you please have a try to confirm? Thanks.

**Comment by xwv** [comment]
No update for a long time. And this might be fixed in swift 3.0 migration.
